_P128_AS3935.ino

```diff
@@ -0,0 +1,558 @@
+    // the whole purpose of interrupts, but in real life you could put your chip to sleep
+    // and lower power consumption or do other nifty things
+
+    success = true;
+  }
+
+  case PLUGIN_TEN_PER_SECOND:
+  {
+    //code to be executed 10 times per second. Tasks which require fast response can be added here
+    //be careful on what is added here. Heavy processing will result in slowing the module down!
+    // Serial.println("jes");
+    byte i2c_addr = Settings.TaskDevicePluginConfig[event->TaskIndex][0];
+
+    if (AS3935IrqTriggered[i2c_addr])
+    {
+      // reset the flag
+      AS3935IrqTriggered[i2c_addr] = false;
+      // first step is to find out what caused interrupt
+      // as soon as we read interrupt cause register, irq pin goes low
+      int irqSource = P128_interruptSource(i2c_addr);
+      UserVar[event->BaseVarIndex + 2] = irqSource;
+      // returned value is bitmap field, bit 0 - noise level too high, bit 2 - disturber detected, and finally bit 3 - lightning!
+      if (irqSource & 0b0001)
+      {
+        addLog(LOG_LEVEL_INFO, F("AS3935   : Noise level too high, try adjusting noise floor"));
+      }
+      else if (irqSource & 0b0100)
+      {
+        addLog(LOG_LEVEL_INFO, F("AS3935   : Disturber detected"));
+      }
+      else if (irqSource & 0b1000)
+      {
+        // need to find how far that lightning stroke, function returns approximate distance in kilometers,
+        // where value 1 represents storm in detector's near victinity, and 63 - very distant, out of range stroke
+        // everything in between is just distance in kilometers
+        int strokeDistance = P128_lightningDistanceKm(i2c_addr);
+        UserVar[event->BaseVarIndex] = strokeDistance;
+        long energy = P128_lightningEnergy(i2c_addr);
+        UserVar[event->BaseVarIndex + 1] = energy;
+        if (strokeDistance == 1)
+          addLog(LOG_LEVEL_INFO, F("AS3935   : Storm overhead, watch out!"));
+        if (strokeDistance == 63)
+          addLog(LOG_LEVEL_INFO, F("AS3935   : Out of range lightning detected"));
+        if (strokeDistance < 63 && strokeDistance > 1)
+          addLog(LOG_LEVEL_INFO, "AS3935   : Lightning detected " + String(strokeDistance) + " kilometers away");
+      }
+      else
+        addLog(LOG_LEVEL_INFO, F("AS3935   : Undefined irq source"));
+
+      sendData(event);
+    }
+    success = true;
+  }
+  } // switch
+  return success;
+
+} //function
+
+//implement plugin specific procedures and functions here
+void p128_irq_0x01()
+{
+  AS3935IrqTriggered[0x01] = true;
+}
+
+void p128_irq_0x02()
+{
+  AS3935IrqTriggered[0x02] = true;
+}
+
+void p128_irq_0x03()
+{
+  AS3935IrqTriggered[0x03] = true;
+}
+
+uint8_t P128_ffsz(uint8_t mask)
+{
+  uint8_t i = 0;
+  if (mask)
+    for (i = 1; ~mask & 1; i++)
+      mask >>= 1;
+  return i;
+}
+
+void P128_registerWrite(uint8_t addr, uint8_t reg, uint8_t mask, uint8_t data)
```

Comment:
Can you also have a look at the `I2C.ino` file and use those functions for I2C communication?
It will make it easier later for me to merge it into the main repository.
And using central functions for low-level tasks makes it easier to fix bugs or use another I2C library.